Controller/PromotionController.js

```diff
@@ -0,0 +1,89 @@
+import InputView from '../Views/InputView.js';
+import Promotion from '../domains/Promotion.js';
+import OutputView from '../Views/OutputView.js';
+import Order from '../domains/Order.js';
+import Discount from '../domains/Discount.js';
+import MESSAGE from '../constants/message.js';
+
+class PromotionController {
+  #promotion;
+
+  async start() {
+    OutputView.printStartMent();
+    await this.getUserPromotion();
+    this.printPromotion();
+  }
+
+  printPromotion() {
+    this.getOrderMenu();
+    this.getTotalPrice();
+    this.getGift();
+    this.getPromotion();
+    this.getTotalDiscount();
+    this.getPriceForPay();
+    this.getEventBadge();
+  }
+
+  async getUserPromotion() {
+    const date = await this.getDateInput();
+    const menu = await this.getOrderInput();
+    OutputView.printVisitDate(date);
+
+    this.#promotion = new Promotion(menu, date);
+  }
+
+  async getOrderInput() {
+    while (true) {
+      const menu = await InputView.readOrder();
+      if (Order.validate(menu)) return menu;
+      OutputView.printError(MESSAGE.order);
+    }
+  }
+
+  async getDateInput() {
+    while (true) {
+      const date = await InputView.readDate();
+      if (Discount.validate(date)) return date;
+      OutputView.printError(MESSAGE.date);
+    }
+  }
+
+  getOrderMenu() {
+    const orderBoard = this.#promotion.readOrderBoard();
+    OutputView.printOrderMenu(orderBoard);
+  }
+
+  getTotalPrice() {
+    let totalPrice = this.#promotion.readTotalPrice();
+    OutputView.printTotalPrcie(totalPrice);
+  }
+
+  getGift() {
+    let gift = MESSAGE.none;
+    if (this.#promotion.checkCanGetGift()) gift = MESSAGE.gift;
```

Comment:
여기 선물을 받아오는 로직을 클래스의 메소드로 만들면 어떨까요?
promotion을 이용해도 좋을 것 같아요!
저 같은 경우엔 선물 교환과 관련된 부분을 별도의 클래스로 만들어서 처리를 해줬어요. 선물 교환하는 부분이 다른 클래스의 일부로 두기에는 꽤 복잡한 것 같다고 생각했었거든요 ㅎㅎ

Reply:
클래스의 메소드로 만들다 라는게 잘 이해가 안됩니다 !!

괜찮으시다면 다시 한번 설명 부탁드려도 될까요 ??

Reply:
넵 물론입니다 :) 다시 보니 제가 좀 모호하게 말씀드린 것 같네요

현재는 promotion 객체의 checkCanGetGift의 메소드로 선물 수령 가능 여부를 받아와서 MESSAGE.gift로 선물 정보를 가져와 선물을 출력하고 있습니다.
그런데 선물 여부에 따라 선물을 반환하는 부분은 비즈니스 로직에 해당하는 것이어서 이 로직을 다른 클래스의 메소드로 처리하면 더 좋을 것 같다고 생각했습니다!
예를 들면 Promotion의 메소드로 옮길 수 있을 것 같습니다.
Promotion의 메소드를 통해 선물을 가져오고 여기 controller에서는 promotion에서 가져온 선물을 출력만 시켜주는 거죠!
그렇게 하면, controller에서 비즈니스 로직을 떼어낼 수 있고, 앞서 PR 코멘트로 언급하신 "객체간의 메세지 교환"에도 더 부합하게 될 것 같습니다.

```suggestion
    // Promotion.js
    getGift() {
         let gift = MESSAGE.none;
         if (this.#promotion.checkCanGetGift()) gift = MESSAGE.gift;
         return gift
    }
    // PromotionController.js
    const gift = this.#promotion.getGift();
```

혹시 이해되지 않는 부분이 있다면 편히 말씀해주세요!

Reply:
충분히 이해했습니다 !!

친절한 답변 정말 감사드립니다 !!

해당 로직도 비즈니스 로직이란걸 당시에는 생각 못햇지만 말씀 들어보니 비즈니스 로직으로 분리했음이 더 좋다는 생각이 듭니다 !!

감사합니다 😊